gym_ignition/base/robot/robot_joints.py

```diff
@@ -0,0 +1,361 @@
+
+        Returns:
+            The joint generalized position.
+
+        """
+
+    @abstractmethod
+    def joint_velocity(self, joint_name: str) -> float:
+        """
+        Return the generalized velocity of the specified joint.
+
+        Args:
+            joint_name: The name of the joint.
+
+        Returns:
+            The joint generalized velocity.
+
+        """
+
+    @abstractmethod
+    def joint_positions(self) -> np.ndarray:
+        """
+        Return the generalized positions of the joints.
+
+        The order matches :meth:`joint_names`.
+
+        Returns:
+            The joint generalized positions.
+        """
+
+    @abstractmethod
+    def joint_velocities(self) -> np.ndarray:
+        """
+        Return the generalized velocities of the joints.
+
+        The order matches :meth:`joint_names`.
+
+        Returns:
+            The joint generalized velocities.
+        """
+
+    @abstractmethod
+    def joint_pid(self, joint_name: str) -> Union[PID, None]:
+        """
+        Return the PID associated to the specified joint.
+
+        Args:
+            joint_name: The joint name.
+
+        Returns:
+            The low-level joint PID or None if no PID was configured.
+        """
+
+    @abstractmethod
+    def dt(self) -> float:
+        """
+        Returns the step_size of the low-level PID controllers.
+
+        Returns:
+            The PIDs step_size.
+        """
+
+    @abstractmethod
+    def set_dt(self, step_size: float) -> bool:
+        """
+        Set the step size used by the low-level PIDs.
+
+        Args:
+            step_size: The PIDs step size.
+
+        Returns:
+            True if successful, False otherwise.
+        """
+
+    @abstractmethod
+    def set_joint_position_target(self, joint_name: str, position_reference: float) -> \
+            bool:
+        """
+        Set the position reference of the specified joint.
+
+        Args:
+            joint_name: The name of the joint.
+            position_reference: The joint position reference.
+
+        Returns:
+            True if successful, False otherwise.
+        """
+
+    @abstractmethod
+    def set_joint_velocity_target(self, joint_name: str, velocity_reference: float) -> \
+            bool:
+        """
+        Set the velocity reference of the specified joint.
+
+        Args:
+            joint_name: The name of the joint.
+            velocity_reference: The joint velocity reference.
+
+        Returns:
+            True if successful, False otherwise.
+        """
+
+    @abstractmethod
+    def set_joint_force(self, joint_name: str, force: float, clip: bool = False) -> bool:
+        """
+        Set the generalized force applied to the specified joint.
+
+        Args:
+            joint_name: The name of the joint.
+            force: The joint generalize force.
+            clip: Clip joint forces reading effort limits.
+
+        Returns:
+            True if successful, False otherwise.
```

Comment:
If the control mode is not velocity control, this returns false?

Reply:
I guess I did something wrong, this comment was meant for the `set_joint_velocity_target` method. I was just asking in which case you got False, as it was  not clear for the docs.

Reply:
In this case I have an inner debate. There are two possibilities:

1. Force the user to set the control mode in advance to make him / her aware of what he's / she's doing (does a design pattern exist to describe something similar? If not, we should make one :))
1. Automatically switch the control mode to velocity when the method is called.

Reply:
Went for 1.

Reply:
Just to let you know: YARP/iCub controlboards used to use 2. and the switched to 1.